overlord/snapstate/handlers_aliasesv2_test.go

```diff
@@ -1443,3 +1441,407 @@ func (s *snapmgrTestSuite) TestDoUndoDisableAliases(c *C) {
 		"alias3": {Manual: "cmd3"},
 	})
 }
+
+func (s *snapmgrTestSuite) TestDoPreferAliasesV2(c *C) {
+	s.state.Lock()
+	defer s.state.Unlock()
+
+	snapstate.Set(s.state, "alias-snap", &snapstate.SnapState{
+		Sequence: []*snap.SideInfo{
+			{RealName: "alias-snap", Revision: snap.R(11)},
+		},
+		Current:             snap.R(11),
+		Active:              true,
+		AutoAliasesDisabled: true,
+		Aliases: map[string]*snapstate.AliasTarget{
+			"alias1": {Auto: "cmd1"},
+			"alias2": {Auto: "cmd2"},
+			"alias3": {Auto: "cmd3"},
+		},
+	})
+	snapstate.Set(s.state, "other-alias-snap1", &snapstate.SnapState{
+		Sequence: []*snap.SideInfo{
+			{RealName: "other-alias-snap1", Revision: snap.R(3)},
+		},
+		Current: snap.R(3),
+		Active:  true,
+		Aliases: map[string]*snapstate.AliasTarget{
+			"alias1": {Auto: "cmd1"},
+		},
+	})
+	snapstate.Set(s.state, "other-alias-snap2", &snapstate.SnapState{
+		Sequence: []*snap.SideInfo{
+			{RealName: "other-alias-snap2", Revision: snap.R(3)},
+		},
+		Current:        snap.R(3),
+		Active:         true,
+		AliasesPending: true,
+		Aliases: map[string]*snapstate.AliasTarget{
+			"alias2": {Manual: "cmd2"},
+			"aliasx": {Manual: "cmdx"},
+		},
+	})
+	snapstate.Set(s.state, "other-alias-snap3", &snapstate.SnapState{
+		Sequence: []*snap.SideInfo{
+			{RealName: "other-alias-snap3", Revision: snap.R(3)},
+		},
+		Current: snap.R(3),
+		Active:  true,
+		Aliases: map[string]*snapstate.AliasTarget{
+			"alias3": {Manual: "cmd5", Auto: "cmd3"},
+		},
+	})
+
+	t := s.state.NewTask("prefer-aliases", "test")
+	t.Set("snap-setup", &snapstate.SnapSetup{
+		SideInfo: &snap.SideInfo{RealName: "alias-snap"},
+	})
+	chg := s.state.NewChange("dummy", "...")
+	chg.AddTask(t)
+
+	s.state.Unlock()
+	for i := 0; i < 3; i++ {
+		s.snapmgr.Ensure()
+		s.snapmgr.Wait()
+	}
+	s.state.Lock()
+	c.Check(t.Status(), Equals, state.DoneStatus, Commentf("%v", chg.Err()))
+
+	// start with an easier-to-read error if this fails:
+	c.Assert(s.fakeBackend.ops.Ops(), DeepEquals, []string{"update-aliases", "update-aliases", "update-aliases"})
+	c.Assert(s.fakeBackend.ops[0].aliases, HasLen, 0)
+	c.Assert(s.fakeBackend.ops[0].rmAliases, HasLen, 1)
```

Comment:
Just curious, why no DeepEqual test here similar to the one below - I guess because there is no order for op[0] and op[1] (?)